### PR TITLE
Feat: whitelist tokenfactory hooks

### PR DESCRIPTION
**Pull request by jcompagni10**
Allow chain manager to update tokenfactory params.

Related PRs:

https://github.com/neutron-org/neutron-dao/pull/107
https://github.com/neutron-org/neutron-sdk/pull/150
https://github.com/neutron-org/neutronjsplus/pull/43
https://github.com/neutron-org/neutron-integration-tests/pull/314

Integration tests:
https://github.com/neutron-org/neutron-tests/actions/runs/9761809686

### CODE REVIEW

File: contracts/dao/neutron-chain-manager/src/contract.rs [hunk 1]
```diff
@@ -266,7 +274,68 @@ fn check_cron_update_msg_params(
...
+    if tokenfactory_params.params.whitelisted_hooks != msg_update_params.params.whitelisted_hooks
+        && !tokenfactory_update_param_permission.whitelisted_hooks
+    {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    Ok(())
+}
+
+/// Queries the parameters of the cron module.
```
Comment (sotnikov-s) [inline contracts/dao/neutron-chain-manager/src/contract.rs:329]:
```suggestion
/// Queries the parameters of the tokenfactory module.
```

File: contracts/dao/neutron-chain-manager/src/testing/tests.rs [hunk 2]
```diff
@@ -643,17 +888,4 @@ pub fn test_execute_execute_unknown_message() {
...
-            title: "0xdeadbeef".to_string(),
-            description: "0xdeadbeef".to_string(),
-            subject_client_id: "0xdeadbeef".to_string(),
-            substitute_client_id: "0xdeadbeef".to_string(),
-        }),
-    });
-
-    let info = mock_info("addr1", &[]);
-    let err = execute_execute_messages(deps.as_mut(), info.clone(), vec![msg]).unwrap_err();
-    assert_eq!(err, Unauthorized {});
```
Comment (sotnikov-s) [inline contracts/dao/neutron-chain-manager/src/testing/tests.rs:658]: why removed?

File: contracts/dao/neutron-chain-manager/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            E: de::Error,
+        {
+            value.parse::<u64>().map_err(de::Error::custom)
+        }
+    }
+
+    deserializer.deserialize_any(StringOrNumberVisitor)
+}
+
+pub fn deserialize_u64_vec<'de, D>(deserializer: D) -> Result<Vec<u64>, D::Error>
```
[the remark this answers is not in the document]
Comment (jcompagni10) [reply contracts/dao/neutron-chain-manager/src/utils.rs:36]: Copied over from this PR https://github.com/neutron-org/neutron-dao/pull/106

But I will just add it there

**Review by swelf19 — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Comment by pr0n00gler** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9794556724

**Pull request merged**